docs/proposals/202106120-edgex-integration.md

```diff
@@ -80,25 +80,235 @@ As the deployment topology diagram shows.  Each components in the EdgeX instance
+            properties:
+              componetsstatus:
+                items:
+                  properties:
+                    deploymentstatus:
+                      description: DeploymentStatus is the most recently observed
+                        status of the Deployment.
+                      type: object
+                  type: object
+                type: array
+              initialized:
+                type: boolean
+              message:
+                description: A human readable message indicating details about the
+                  transition.
+                type: string
+              reason:
+                description: The reason for the condition's last transition.
+                type: string
+            type: object
+        type: object
+    served: true
+    storage: true
+    subresources:
+      status: {}
+status:
+  acceptedNames:
+    kind: ""
+    plural: ""
+  conditions: []
+  storedVersions: []
 
 ```
 
 
 ```go
+type PoolSpec struct {
+	// Indicates pool name as a DNS_LABEL, which will be used to generate
+	// pool workload name prefix in the format '<deployment-name>-<pool-name>-'.
+	// Name should be unique between all of the pools under one UnitedDeployment.
+	// Name is NodePool Name
+	Name string `json:"name"`
+
+	// Indicates the node selector to form the pool. Depending on the node selector,
+	// pods provisioned could be distributed across multiple groups of nodes.
+	// A pool's nodeSelectorTerm is not allowed to be updated.
+	// +optional
+	NodeSelectorTerm corev1.NodeSelectorTerm `json:"nodeSelectorTerm,omitempty"`
+
+	// Indicates the tolerations the pods under this pool have.
+	// A pool's tolerations is not allowed to be updated.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Indicates the number of the pod to be created under this pool.
+	// +required
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Indicates the patch for the templateSpec
+	// Now support strategic merge path :https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/#notes-on-the-strategic-merge-patch
+	// Patch takes precedence over Replicas fields
+	// If the Patch also modifies the Replicas, use the Replicas value in the Patch
+	// +optional
+	Patch *runtime.RawExtension `json:"patch,omitempty"`
+}
+
+// DeploymentTemplateSpec defines the pool template of Deployment.
+type DeploymentTemplateSpec struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              appsv1.DeploymentSpec `json:"spec"`
+}
+
+// DeploymentTemplateSpec defines the pool template of Deployment.
+type ServiceTemplateSpec struct {
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              appsv1.DeploymentSpec `json:"spec"`
+}
+
+type ComponetSpec struct {
+	// +optional
+	Deployment DeploymentTemplateSpec `json:"deploymentspec,omitempty"`
+
+	// +optional
+	Service ServiceTemplateSpec `json:"servicespec,omitempty"`
```

Comment:
I think that all CRs for one kind Deployment will share the same service, so maybe it's not suitable to define Service in `ComponentSpec`.